aten/src/ATen/native/sparse/SparseTensorMath.cpp

```diff
@@ -508,7 +541,13 @@ SparseTensor& s_mul_out_sparse_cpu(SparseTensor& r, const SparseTensor& t_, cons
     );
   }
 
+#ifndef USE_TH_SIZE_ZERO_DIM
   _get_sparse_impl(r)->set_nnz(r_i);
+#else
+  r_indices = r_indices.narrow(1, 0, r_i);
+  r_values = r_values.narrow(0, 0, r_i);
+  _get_sparse_impl(r)->set_indices_and_values_unsafe(r_indices, r_values);
```

Comment:
We actually still need `set_nnz` in the existing code because `narrow` doesn't support 0-length without the flag yet, so we can't reorganize the existing code to use the new logic.

Reply:
I get that part; I was referring to `set_indices_and_values_unsafe` -- I don't see any reason those can't be in the same place in both the ifdef and non-ifdef cases.

Reply:
I think it actually might be a bit tricky here -- `set_indices_and_values_unsafe` would actually sets the `nnz` to `values.size(0)`, so in the existing code that function should be called before `set_nnz`, but in the new code it should be called after `r_indices = r_indices.narrow(1, 0, r_i)` and `r_values = r_values.narrow(0, 0, r_i)` so that it has the right indices and values. We could probably write it as:
```
#ifdef USE_TH_SIZE_ZERO_DIM
  r_indices = r_indices.narrow(1, 0, r_i);
  r_values = r_values.narrow(0, 0, r_i);
#endif

_get_sparse_impl(r)->set_indices_and_values_unsafe(r_indices, r_values);

#ifndef USE_TH_SIZE_ZERO_DIM
  _get_sparse_impl(r)->set_nnz(r_i);
#endif
```